### PR TITLE
Added check - check for promise

### PR DESCRIPTION
**Pull request by Jayasankar-m**
Check for promise
eg: is.promise(obj)  - returns true if promise

### CODE REVIEW

File: src/is/types.js [hunk 1]
```diff
@@ -158,6 +158,15 @@ export function isWindowObject(value) {
     return value != null && typeof value === 'object' && 'setInterval' in value;
 }
 
+/**
+ * Determines whether a given value is a promise
+ * @param {*} value the value to check
+ * @returns {Boolean} `true` if the value is a promise; otherwise, `false`
+ */
+export function isPromise(value) {
+    return typeof value.then === 'function';
```
Comment (joeltankam) [inline src/is/types.js]: This is not a [good specification](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) for promises. `{ then : function(){} }` is a regular object with a `then` function, but is considered as a promise by your implementation. 
Also, you should use `is.function` for this kind of verification.

File: src/is/types.js [hunk 2]
```diff
@@ -188,4 +197,5 @@ export default {
     dom: isDomNode,
     windowObject: isWindowObject,
     window: isWindowObject,
+    promise: isPromise,
```
Comment (joeltankam) [inline src/is/types.js]: Reorder below `isFunction`

**Review by joeltankam — CHANGES_REQUESTED** [review]
Thanks for you contribution. 
Some changes needed. Also, please implement tests and update documentation

File: src/is/types.js (hunk 1 above)
Comment (Jayasankar-m) [reply src/is/types.js]: Thanks for the update.
How about using `toString.call(value) === '[object Promise]';`

Shall i update with this?

**Pull request closed without merging**